### PR TITLE
Oup parsing journal doctype

### PR DESCRIPTION
**Pull request by ErnestaP**

### CODE REVIEW

**Comment by ErnestaP** [comment]
Will be ready for review, after https://github.com/cern-sis/workflows/pull/102 is merged, for less confusion

File: dags/oup/parser.py [hunk 1]
```diff
@@ -34,3 +39,19 @@ def _get_dois(self, article: ET.Element):
...
+    def _get_journal_doctype(self, article: ET.Element):
+        self.logger.msg("Parsing journal doctype for article", dois=self.dois)
+        journal_doctype_raw = article.find("front/..").get("article-type")
+        if not journal_doctype_raw:
+            return
+        try:
+            journal_doctype = self.article_type_mapping[journal_doctype_raw]
+        except KeyError:
+            journal_doctype = "other"
+        if "other" in journal_doctype:
```
Comment (MJedr) [inline dags/oup/parser.py]: does it happen anytime except of the `KeyError`? If no, maybe you can remove `if` statement

File: tests/units/oup/test_oup_parser.py [hunk 2]
```diff
@@ -36,14 +35,40 @@ def test_dois(parsed_articles):
...
     article_name = "ptac108_without_doi_value.xml"
     with open(shared_datadir / article_name) as file:
-        content = ET.fromstring(file.read())
+        content = parse_without_names_spaces(file.read())
         with raises(RequiredFieldNotFoundExtractionError):
             parser._publisher_specific_parsing(content)
+
+
+def test_journal_doc_types(parsed_articles):
+    doc_types = ["article", "article", "article"]
```
Comment (MJedr) [inline tests/units/oup/test_oup_parser.py]: would be nice to have an article in other type there

File: dags/oup/parser.py (hunk 1 above)
Comment (ErnestaP) [reply dags/oup/parser.py]: Yes, just when the type is not found in article type mapping

File: tests/units/oup/test_oup_parser.py (hunk 2 above)
Comment (ErnestaP) [reply tests/units/oup/test_oup_parser.py]: I can try to find one

Comment (ErnestaP) [reply tests/units/oup/test_oup_parser.py]: managed to find just 1

File: dags/oup/parser.py (hunk 1 above)
Comment (ErnestaP) [reply dags/oup/parser.py]: Can it be a different exception in this case?

Comment (MJedr) [reply dags/oup/parser.py]: no, I would just move logger warning to the exception handling and remove `if` clause

**Review by MJedr — APPROVED** [review]

**Pull request merged**